iceoryx_posh/source/runtime/posh_runtime.cpp

```diff
@@ -418,13 +419,22 @@ cxx::expected<Error> PoshRuntime::findService(const capro::ServiceDescription& s
         errorHandler(Error::kPOSH__SERVICE_DISCOVERY_INSTANCE_CONTAINER_OVERFLOW, nullptr, ErrorLevel::MODERATE);
         return cxx::error<Error>(Error::kPOSH__SERVICE_DISCOVERY_INSTANCE_CONTAINER_OVERFLOW);
     }
-    return {cxx::success<>()};
+    return {cxx::success<InstanceContainer>(instanceContainer)};
 }
 
-void PoshRuntime::offerService(const capro::ServiceDescription& serviceDescription) noexcept
+
+bool PoshRuntime::offerService(const capro::ServiceDescription& serviceDescription) noexcept
 {
-    capro::CaproMessage msg(capro::CaproMessageType::OFFER, serviceDescription, capro::CaproMessageSubType::SERVICE);
-    m_applicationPort.dispatchCaProMessage(msg);
+    if (serviceDescription.isValid())
+    {
+        capro::CaproMessage msg(
+            capro::CaproMessageType::OFFER, serviceDescription, capro::CaproMessageSubType::SERVICE);
+        m_applicationPort.dispatchCaProMessage(msg);
+        return true;
+    }
+    LogWarn() << "Could not offer service " << serviceDescription.getServiceIDString() << ","
+              << " ServiceDescription is invalid\n";
+    return false;
```

Comment:
Can you please extend the tests to the new added functionality and resolve the codecov warnings?

Reply:
Tests are already added for newly added functionality, but that is under integration tests not module